TypeCobol/Compiler/Nodes/Data.cs

```diff
@@ -282,31 +283,193 @@ public virtual DataType PrimitiveDataType
+                            occupiedMemory += ((DataDefinition)parent.Children[i]).SlackBytes;
+                        }
+
+                        index = parent.Parent.ChildIndex(parent);
+                        parent = parent.Parent as DataDefinition;
+                    }
+
+                    
+                    int m = 1;
+
+                    switch (Usage.Value)
+                    {
+                        case DataUsage.Binary:
+                            Types.PictureValidator pv = new Types.PictureValidator(this.Picture.Value, SignIsSeparate);
+                            pv.IsValid();
+                            if (pv.ValidationContext.Digits <= 4)
+                            {
+                                m = 2;
+                            }
+                            else
+                            {
+                                m = 4;
+                            }
+                            break;
+                        case DataUsage.Index:
+                        case DataUsage.Pointer:
+                        case DataUsage.ProcedurePointer:
+                        case DataUsage.ObjectReference:
+                        case DataUsage.FunctionPointer:
+                        case DataUsage.PackedDecimal:
+                            m = 4; 
+                            break;
+                        case DataUsage.FloatingPoint:
+                            m = 8;
+                            break;
+                    }
+
+                    if (occupiedMemory % m > 0)
+                    {
+                        _slackBytes = m - (occupiedMemory % m);
+                    }
+
+
+                }
+
+                return _slackBytes.Value;
+
+            }
+        }
+
+        private long? _startPosition = null;
+        public virtual long StartPosition
+        {
+            get
+            {
+                if (_startPosition.HasValue)
+                {
+                    return _startPosition.Value;
+                }
+
+                if (this is DataRedefines)
+                {
+                    SymbolReference redefined = (CodeElement as DataRedefinesEntry)?.RedefinesDataName;
+                    var results = SymbolTable.DataEntries.First(de => de.Key == redefined?.Name).Value;
+
+                    if (results.Count == 1)
+                    {
+                        _startPosition = results.First().StartPosition;
+                        return _startPosition.Value;
+                    }
+                    
+                }
+
+                if (Parent is DataSection)
+                {
+                    _startPosition = 1;
                 }
                 else
-                    return 0;
+                {
+                    for (int i = 0; i < Parent.Children.Count; i++)
+                    {
+                        DataDefinition sibling = (DataDefinition)Parent.Children[i];
```

Comment:
You can declare sibling as Node (or use `var`), then cast to DataDefinition only on line 457.
This will reduce the number of cast